src/scaladoc/scala/tools/nsc/ScalaDoc.scala

```diff
@@ -19,13 +19,15 @@ class ScalaDoc {
 
   def process(args: Array[String]): Boolean = {
     var reporter: ScalaDocReporter = null
-    val docSettings = new doc.Settings(msg => reporter.error(FakePos("scaladoc"), msg + "\n  scaladoc -help  gives more information"),
+    val docSettings = new doc.Settings(msg => reporter.error(FakePos("scaladoc"), msg + "\n  scaladoc -help gives more information"),
                                        msg => reporter.printMessage(msg))
     reporter = new ScalaDocReporter(docSettings)
     val command = new ScalaDoc.Command(args.toList, docSettings)
     def hasFiles = command.files.nonEmpty || docSettings.uncompilableFiles.nonEmpty
 
-    if (docSettings.version.value)
+    if (!command.ok)
+      return false; // hard exit if there are errors, e.g. bad arguments
```

Comment:
Can this follow the style of the other branches?

- Don't use `return`
- Use `reporter`